src/libraries/Common/src/System/Globalization/FormatProvider.Number.cs

```diff
@@ -517,15 +517,19 @@ private static unsafe bool ParseNumber(ref char* str, char* strEnd, NumberStyles
                             {
                                 exp = exp * 10 + (ch - '0');
                                 ch = ++p < strEnd ? *p : '\0';
-                                if (exp > 1000)
-                                {
-                                    exp = 9999;
-                                    while (char.IsAsciiDigit(ch))
-                                    {
-                                        ch = ++p < strEnd ? *p : '\0';
-                                    }
-                                }
-                            } while (char.IsAsciiDigit(ch));
+                            } while (char.IsAsciiDigit(ch) && (exp < int.MaxValue / 10));
```

Comment:
Would it be better to limit this to `999,999,999` like we did for `ToString("G#########")`?

Reply:
It might be safter in case there is any incrementing that happens elsewhere that I hadn't noticed. Would that require updating documentation somewhere?

Reply:
Likely a similar breaking change doc to what we did for `ToString`.